### ISSUE TITLE
Quality check for when a user completes a mission with zero labels

### ISSUE THREAD

**Issue by maddalihanumateja** [ProjectSidewalk/SidewalkWebpage#1082]
This applies to cases when the total label count for a missions turns out to be zero. Ask for a manual confirmation on whether the user really didnt see any issues during the mission. Also, possibly mark the street edge for re-auditing by another Turker and dont include the street edges associated with that mission for the landing page stats

**Comment by jonfroehlich** [comment]
Thanks @maddalihanumateja for creating this issue. For context, on Sept 13, we started to notice that one or more users were completing audits with very few (or zero) labels. This should almost never be the case. I don't know the calculation, but I think an average audit is something like ~50 labels or something.

Yesterday, we had 82 audits but only 164 labels (that's 2 labels/audit on average) and today we have 116 audits with only 18 labels (0.15 labels per audit). 

![image](https://user-images.githubusercontent.com/1621749/30392987-5c154248-9873-11e7-968e-33a79f145101.png)

So, it looks like we are seeing some vandalism on the site and this means that we need to have simple checks (which we can make more sophisticated in the future) to: 

- mark these audit routes for re-audit
- proactively intervene with interactive dialogs that tell the user we have noticed that they are not auditing in the way we expecting (we'd have to work on the phrasing and the way to do this)
- possibly prevent users from further auditing (in worst case)

**Comment by misaugstad** [comment]
This is also discussed in #615

**Comment by misaugstad** [comment]
I'm sure that we will start with an MVP, but for the future... I was just thinking that, as an input to some algorithm we write that tries to estimate the quality of an audit, we could use something like the distance of labels (of certain label types) from intersections.

The most obvious case would be if we were seeing curb ramps and missing curb ramps that are far from intersections, the audit might be marked as low quality. Obviously this should be verified using the GT that we now have, but I see it as another thing we could factor into a more sophisticated algorithm for getting high-quality audits.

**Comment by misaugstad** [comment]
So I see two similar, though slightly different problems here (at least in practice). There is filtering out work we suspect to be of low quality based on some algorithm/metrics, and then there is filtering out work that is deliberately done in bad faith. And I would like to be able to manually add users to the latter list.

So the workers that we determine to be auditing in bad faith (either by manual review, or algorithmically if we use an extreme threshold), we should just filter those labels/audits out from just about everything. We don't want to include their labels in the label count on our landing page, we don't want to return their labels via the API, and we don't want to include their labels when creating visualizations of our data.

Workers that we suspect may have low quality work based on some algorithm/metrics is a different story. We probably want to include their data in most of those places. When doing analysis or visualization, we may want to apply less weight to their labels, but we don't just want to completely disregard them at all times.

Another difference I see is that a user may be placed on the list of "possibly low quality work", and may then be pulled off the list later on, depending on how our model changes, or on how their labeling behavior changes as they continue. But for the list of workers auditing in bad faith, I don't really expect users to be removed from that list much at all.


I'm wondering if it would be a good idea to create some table in the database (or a column in the user table) that maintains a list of users we have verified as auditing in bad faith. I am building up such a list as I review HITs that we post to mturk, and I would like to do something with this list. Especially considering our model, which is very simple right now and would not filter out all such bad-faith users.

**Comment by jonfroehlich** [comment]
## Revisiting how we infer user/label quality (a signal roadmap)

Picking this thread back up while working on #3901. @misaugstad's framing above still holds and maps cleanly onto what we built since: the **"bad faith"** list became `user_stat.excluded` (hard-filtered everywhere), and the **"possibly low quality"** list became `user_stat.high_quality` (weighted/hidden but recoverable). The wish for "a more sophisticated algorithm" is the part still mostly open — so here's a structured menu of quality signals, ordered by cost, to anchor that.

### Where we are today
`user_stat` infers quality from **two earned signals** collapsed into one boolean:
- **`accuracy`** = consensus agreement on the user's own labels (`Σcorrect / (Σcorrect + Σincorrect)`), only "counted" once `own_labels_validated ≥ 50`;
- **`labels_per_meter`** = labeling frequency.

`high_quality` is false if *either* is low. That conflation is exactly the bug in #3901 (a sparse-but-accurate user gets hidden), and it's the root limitation: **one per-user boolean from two signals.**

### A. Squeeze more from data we already have (arithmetic, no new collection)
1. **Confidence-weighted accuracy instead of the 50-label cliff.** We have `n` (`own_labels_validated`) and `p` (`accuracy`); a **Wilson/Beta lower bound** gives a smooth, fair estimate — 65% over 500 labels scores well, 65% over 8 stays cautious. This replaces a magic threshold with a principled one and directly fixes the #3901 "what about unproven sparse users?" edge case.
2. **Per-label-type accuracy.** Accuracy is one scalar today, but a user can be reliable on curb ramps and weak on surface problems. We have every (label, validation) pair to compute accuracy per *(user, label_type)*. Probably the **single biggest accuracy gain**, fully derivable — it just means quality stops being one number.
3. **Recency weighting.** Users improve; decaying old labels stops us from judging someone on who they were two years ago.

### B. Signals we have in other tables but `user_stat` ignores
4. **Corroboration via clustering.** The clustering job already knows which labels co-locate. A label independently re-found by another user (same feature/type, nearby) is strong evidence of correctness — with **no explicit validation needed**, sidestepping the validation backlog. (Relatedly, #2478 / @misaugstad's label-distance-from-intersection idea above is the same family: geometry-as-quality-signal.)
5. **AI concurrence.** AI validations are now baked into `agree/disagree`. Agreement-with-AI is a fast, scalable signal — especially to *bootstrap* a quality estimate before 50 human validations exist (cold-start).
6. **Validator reliability ≠ labeler quality.** Every user does two jobs. We can measure how often someone agrees with consensus when *validating*, and weight their votes accordingly — better consensus → better accuracy estimates for everyone. (This is #2755.)
7. **Behavioral / interaction priors.** We log clicks, pano pans, timestamps, tutorial attempts. Time-on-task (suspiciously fast = careless), tutorial performance, and pan behavior are weak alone but useful **priors for new users**. (Labeling-frequency-as-signal is its own open question in #2753.)

### C. Methodological upgrades (research-grade)
8. **Latent worker-quality models (Dawid–Skene / EM).** The principled version of #2 + #6, and the crowdsourcing-QC standard. Today's accuracy = agreement-with-majority, which is **circular** — the "truth" is itself crowd-derived from the users we're grading. Dawid–Skene jointly estimates label truth *and* each worker's per-class confusion matrix, handling both circularity and per-type heterogeneity.
9. **Gold / honeypot tasks.** Seed known-answer panos (known-good *and* known-bad) into the label/validate streams to measure accuracy **directly and fast**, without waiting for organic validations. Solves cold-start and gaming; needs a curated gold set + plumbing. (This is the modern version of the "GT missions" idea in #1446.)
10. **Low-effort / spammer detection.** Behavioral entropy (always the same label type, always "agree", straight-line placement) catches adversaries that accuracy alone misses — the descendant of the zero-label-mission check that started this issue.

### Cross-cutting themes
- **One boolean is lossy.** The real arc is per-*user* flag → per-*(user, label-type)* → per-*label* confidence. #3901's accuracy/frequency split is step 1 of that, not a one-off patch.
- **Circularity is the deep issue.** #4 (corroboration), #8 (latent models), and #9 (gold) are the three ways out of grading the crowd by the crowd.
- **Cold-start & fairness.** Hard cutoffs are cliffs; #1 (Wilson), #5 (AI prior), #7 (behavioral prior), #9 (gold) all make new-user judgments smoother — and answer "do we surface unproven users?" with math rather than a coin flip.

### Suggested path
- **Near-term (inside #3901):** do the small correct thing, but fold in a **Wilson lower bound (#1)** rather than a hard "accuracy ≥ 60% OR < 50 validations" rule. Same scope, principled edge case, test-friendly.
- **Roadmap:** #2 (per-type accuracy), #4 (corroboration), #8 (Dawid–Skene) are the high-leverage strategic moves; #9 (gold tasks) is the big infrastructure bet.

Related: #3901 (the immediate trigger), #2753 (frequency as a signal), #2755 (validator quality), #2507 (stale results mid-clustering), #2478 (geometry as a signal).

_(Captured from a design discussion; posting here to keep the QC-mechanisms conversation in one place.)_

**Comment by misaugstad** [comment]
One thing worth noting since we've had our current filters up for a few years: For our partners that run deployments, It can be unclear whether/how data is filtered out. For example, labels from users who are marked as low quality are filtered out from LabelMap (unless they've been validated as correct), and I have had a number of instances where partners have come to me to ask where the labels are for some users. And the answer is always that the user was marked as "low quality" by our heuristics (and I usually agree with that assessment).

In some ways that's a separate issue. But it would be really helpful if our metrics for filtering out work are easily explainable to our partners!

**Comment by jonfroehlich** [comment]
Yes, it could even come in as a definition in the JSON itself--though often our partners use CSV, etc. too.
